src/plugin/kubearmor.go

```diff
@@ -404,7 +404,9 @@ func StartKubeArmorRelay(StopChan chan struct{}, cfg types.ConfigKubeArmorRelay)
 		defer func() {
 			log.Info().Msg("watchlogs returning")
 			KubeArmorRelayStarted = false
-			_ = conn.Close()
+			if conn != nil {
+				_ = conn.Close()
+			}
```

Comment:
instead of checking the nil pointer here, it would be better if we can have it while calling the function:
```
	conn := ConnectKubeArmorRelay(cfg)
	if conn == nil {
		// handle error
	}
```